test/functional/README.md

```diff
@@ -51,10 +51,13 @@ don't have test cases for.
 
 #### General test-writing advice
 
+- Instead of inline comments or no test documentation at all, prefer to log the comments to the test log. E.g.
```

Comment:
- maybe omit "prefer to"

- s/. E.g./, e.g./

Reply:
Done both